source/io/module_javaScriptInvoke.js

```diff
@@ -213,7 +213,14 @@
                 isValidReturnType: function (value) {
                     return value instanceof Float64Array;
                 }
-            }
+            },
+			ObjectRefNum: {
+                reader: Module.eggShell.dataReadObjectRefNum,
+                writer: Module.eggShell.dataWriteObjectRefNum,
+                isValidReturnType: function (value) {
+                    return typeof value === 'object';
```

Comment:
Don't think we should validate the value here. The user could pass null, undefined, etc. Maybe just always return true